modules/swagger-codegen/src/main/resources/qt5cpp/api-body.mustache

```diff
@@ -185,9 +185,9 @@ void
     QJsonObject obj = doc.object();
 
     foreach(QString key, obj.keys()) {
-        qint32* val;
```

Comment:
Should be {{{returnBaseType}}}

Reply:
Not part of this PR, but in case the {{{returnBaseType}}} is dynamically allocated (Object, QString,QDateTime,QbyteArray ... etc ), it has to be wrapped with QObjectWrapper for later deletion.

Reply:
Could you tell me more about QObjectWrapper? I never used it.

Reply:
Ahh, it is a template class which holds the pointer to the memory allocated for dynamically allocated types like QString, Object, QByteArray etc.., for the parameter to be passed in the slot callback. In this file you can find it being used, so once the signal is emitted and the slot activated, the deleteLater ensures the object is deleted after the process event has finished.
Present in the template QObjectWrapper.mustache

auto objwrapper = new {{prefix}}QObjectWrapper<{{{returnBaseType}}}*> (o);
objwrapper->deleteLater();

Reply:
Like I mentioned, it is not part of the crash, but simply to avoid memory leaks for complex types. If you have the time it is a good thing to add, to remove memory leaks. But for the qt5cpp petstore example the cleanup is not working because the loop is exited on receiving the callback. In a real qt application the event loop would still be running and the object deleted.